operator/pkg/ciliumendpointslice/endpointslice_test.go

```diff
@@ -54,240 +52,131 @@ func createStoreEndpointSlice(name string, namespace string, endpoints []capi_v2
-	for _, id := range ids {
-		cep := capi_v2a1.CoreCiliumEndpoint{IdentityID: id}
-		ces.Endpoints = append(ces.Endpoints, cep)
+func TestSyncCESsInLocalCache(t *testing.T) {
+	var r *reconciler
+	var fakeClient k8sClient.FakeClientset
+	m := newCESManagerFcfs(2, log).(*cesManagerFcfs)
+	var ciliumEndpoint resource.Resource[*cilium_v2.CiliumEndpoint]
+	var ciliumEndpointSlice resource.Resource[*cilium_v2a1.CiliumEndpointSlice]
+	hive := hive.New(
+		k8sClient.FakeClientCell,
+		k8s.ResourcesCell,
+		cell.Invoke(func(c *k8sClient.FakeClientset, cep resource.Resource[*cilium_v2.CiliumEndpoint], ces resource.Resource[*cilium_v2a1.CiliumEndpointSlice]) error {
+			fakeClient = *c
+			ciliumEndpoint = cep
+			ciliumEndpointSlice = ces
+			return nil
+		}),
+	)
+	hive.Start(context.Background())
+	r = newReconciler(fakeClient.CiliumFakeClientset.CiliumV2alpha1(), m, log, ciliumEndpoint, ciliumEndpointSlice)
+	cesStore, _ := ciliumEndpointSlice.Store(context.Background())
+	cesController := &Controller{
+		logger:                         log,
+		clientset:                      fakeClient.Clientset,
+		ciliumEndpoint:                 ciliumEndpoint,
+		ciliumEndpointSlice:            ciliumEndpointSlice,
+		reconciler:                     r,
+		manager:                        m,
+		writeQPSBurst:                  2,
+		writeQPSLimit:                  1,
+		enqueuedAt:                     make(map[CESName]time.Time),
+		preInitEnqueuedEndpointsEvents: make([]EndpointEvent, 0),
+		endpointsMappingInitialized:    false,
 	}
-	return ces
+	cesController.initializeQueue()
+
+	cep1 := createManagerEndpoint("cep1", 1)
+	cep2 := createManagerEndpoint("cep2", 1)
+	cep3 := createManagerEndpoint("cep3", 2)
+	cep4 := createManagerEndpoint("cep4", 2)
+	ces1 := createStoreEndpointSlice("ces1", "ns", []cilium_v2a1.CoreCiliumEndpoint{cep1, cep2, cep3, cep4})
+	cesStore.CacheStore().Add(ces1)
+	cep5 := createManagerEndpoint("cep5", 1)
+	cep6 := createManagerEndpoint("cep6", 1)
+	cep7 := createManagerEndpoint("cep7", 2)
+	ces2 := createStoreEndpointSlice("ces2", "ns", []cilium_v2a1.CoreCiliumEndpoint{cep5, cep6, cep7})
+	cesStore.CacheStore().Add(ces2)
+
+	cesController.syncCESsInLocalCache(context.Background())
+
+	mapping := m.mapping
+
+	cesN, _ := mapping.getCESName(NewCEPName("cep1", "ns"))
+	assert.Equal(t, cesN, NewCESName("ces1"))
+	cesN, _ = mapping.getCESName(NewCEPName("cep2", "ns"))
+	assert.Equal(t, cesN, NewCESName("ces1"))
+	cesN, _ = mapping.getCESName(NewCEPName("cep3", "ns"))
+	assert.Equal(t, cesN, NewCESName("ces1"))
+	cesN, _ = mapping.getCESName(NewCEPName("cep4", "ns"))
+	assert.Equal(t, cesN, NewCESName("ces1"))
+	cesN, _ = mapping.getCESName(NewCEPName("cep5", "ns"))
+	assert.Equal(t, cesN, NewCESName("ces2"))
+	cesN, _ = mapping.getCESName(NewCEPName("cep6", "ns"))
+	assert.Equal(t, cesN, NewCESName("ces2"))
+	cesN, _ = mapping.getCESName(NewCEPName("cep7", "ns"))
+	assert.Equal(t, cesN, NewCESName("ces2"))
```

Comment:
Fixed stopping the queue.

Regarding IgnoreCurrent - Shouldn't we always use it?
Otherwise it's easy to add a test for A (e.g. some CES controller test leaking) and change some other part B (e.g. Identity GC) that does have such test and the test for B would start failing even though the problem is totally outside of B.

Reply:
That's true but, if we keep that, it's easier to add a new test that might leak a goroutine, and let the leak go unnoticed. I find this to be even worse, although you're right that having a failure in another test is not ideal.
The hive based tests are like integration tests for multiple cells even if they are written as unit. I suggest to first add goleak to each hive based test (like the ones in `endpointslice_test.go` and in `reconciler_test.go`) and then check that we are not leaking anything (except for anything that is due to a dependency like client-go). If so, I still suggest to avoid `IgnoreCurrent` for now.

Reply:
Also - those test would fail if some other test fail and doesn't clean up everything.
```
hive.Start()

err := checkSth()
if err != nil {
  t.Fatalf("failed")
}

hive.Stop()
```

Removed IgnoreCurrent for now.

It would be good to have some kind of a leak test at the end if all other test passed, maybe?

Reply:
> Also - those test would fail if some other test fail and doesn't clean up everything.

That's true but that is less harmful, since it'll be clear that the root cause is the first test failure.

> It would be good to have some kind of a leak test at the end if all other test passed, maybe?

I think that right now is not necessary, since we can check every single hive-based test with goleak, excluding what we already know is leaking (and out of our control), like `k8s.io/client-go/util/workqueue.(*Type).updateUnfinishedWorkLoop`.